docs/reference/settings.md

```diff
@@ -59,6 +59,46 @@ Linux, and `%LOCALAPPDATA%\uv\cache` on Windows.
 
 ---
 
+#### [`cache-keys`](#cache-keys) {: #cache-keys }
+
+The keys to consider when caching builds for the project.
+
+By default, uv will rebuild a project whenever the `pyproject.toml`, `setup.py`, or
+`setup.cfg` files in the project directory are modified. Cache keys enable you to specify
+additional files or directories that should trigger a rebuild when modified.
+
+For example, if a project uses dynamic metadata to read its dependencies from a
+`requirements.txt` file, you can specify `cache-keys = [{ file = "requirements.txt" }]` to
+ensure that the project is rebuilt whenever the `requirements.txt` file is modified.
+
+Cache keys can also include version control information. For example, if a project uses
+`setuptools_scm` to read its version from a Git tag, you can specify
+`cache-keys = [{ git = true }]` to include the current Git commit hash in the cache key.
+
+Cache keys only affect the project defined by the `pyproject.toml` in which they're
+specified (as opposed to, e.g., affecting all members in a workspace).
+
+**Default value**: `[]`
+
+**Type**: `list[dict]`
+
+**Example usage**:
+
+=== "pyproject.toml"
+
+    ```toml
+    [tool.uv]
+    cache-keys = [{ file = "requirements.txt" }, { git = true }]
```

Comment:
I don't feel strongly about this, but we don't use `extend-` at all in uv yet and you won't be able to change the base cache keys so I don't think it's necessary here.

Reply:
I'm not particular on the specific naming, but i think it's an odd design that we expose a `cache-keys` setting that then doesn't let you configure the cache keys, but add files/git to our immutable defaults.

Reply:
What if we make our current settings a default, and this overrides them?

Reply:
I'd find that more intuitive. It's a bit annoying that you have to re-specify `pyproject.toml` and/or `setup.py`, which i assumed that's what you wanted to avoid with that design; but personally i like just specifying all relevant files.

Reply:
Certainly an option, but are there cases where you would actually want to ignore changes to those other files?